src/main/java/com/commercetools/sync/commons/models/NonResolvedReferencesCustomObject.java

```diff
@@ -0,0 +1,56 @@
+package com.commercetools.sync.commons.models;
+
+import io.sphere.sdk.products.ProductDraft;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class NonResolvedReferencesCustomObject {
+
+    private String productKey;
+
+    private ProductDraft productDraft;
+
+    private List<String> dependantProductKeys;
+
+    /**
+     * Object represent the value for the custom object that will hold unresolved references
+     * @param productKey product key that has non-resolved references
+     * @param productDraft product draft which has non-resolved references
+     * @param dependantProductKeys product keys of non-resolved references
+     */
+    public NonResolvedReferencesCustomObject(@Nonnull final String productKey, @Nonnull final ProductDraft productDraft,
+                                             final List<String> dependantProductKeys) {
+        this.productKey = productKey;
+        this.productDraft = productDraft;
+        this.dependantProductKeys = dependantProductKeys;
+    }
+
+    public NonResolvedReferencesCustomObject() {
+    }
+
+    public String getProductKey() {
+        return productKey;
+    }
+
+    public void setProductKey(final String productKey) {
```

Comment:
`@Nonnull`

Reply:
We can ignore this; as we have already removed this particular variable